### ISSUE TITLE
Setting typing delay to 0 means bot never responds

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#517]
# Description
When a typing delay is defined it is an integer which states how many characters per second opsdroid should type. If you set this to `0` then it can type no characters per second and therefore never responds.

## Steps to Reproduce
- Configure a connector with a `typing-delay` of `0`. 
- Talk to the bot

## Experienced Functionality
The bot never responds.

## Expected Functionality
I would expect the bot to respond eventually. Perhaps even immediately and log an error stating `0` is an invalid response.

Thinking a little more about this I wonder if this functionality is the wrong way round. With `thinking-delay` that is the number of seconds that opsdroid should way before responding, so as the number gets bigger the longer opsdroid waits. But with `typing-delay` it is the number of characters per second that opsdroid can type, so the bigger the number the less opsdroid waits. The word `delay` suggests that the higher the number the longer the wait.

These opposites could be confusing (it confused me this evening). I think it would be good that both numbers represent time, rather than one representing characters per second. That would involve changing it to be seconds per character. This would result in a bigger number causing a bigger delay. 

## Versions
- **Opsdroid version:** master
- **Python version:** 3.5.4
- **OS/Docker version:** macOS 10.13

## Configuration File

```yaml
connectors:
  - name: shell
    typing-delay: 0

skills:
  - name: hello
```

**Comment by jacobtomlinson** [comment]
@FabioRosado I would be interested in your thoughts as you implemented this

**Comment by FabioRosado** [comment]
I agree that both should be measured in seconds to avoid confusion. I can't remember why I decided to go with characters per second to be honest, perhaps it was to divide these two functions because otherwise one would be enough (saying that should we refactor the code to just use one delaying function?)

 I also remember setting a default on that delay but for some reason I didn't added it in the final PR - my bad! I'm happy to fix this issue and change the code if we decide to go with just one function that can take input of the two configuration parameters.
